File: src/main/java/plus/maa/backend/controller/response/user/CopilotSetListRes.java
package plus.maa.backend.controller.response.user;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * @author dragove
 * create on 2024-01-06
 */
@Getter
@Setter
@Schema(title = "作业集响应（列表）")
public class CopilotSetListRes {

    @Schema(title = "作业集id")
    private long id;

    @Schema(title = "作业集名称")
    private String name;

    @Schema(title = "额外描述")
    private String description;

    @Schema(title = "上传者id")
    private String creatorId;

    @Schema(title = "上传者昵称")
    private String creator;

    @Schema(title = "创建时间")
    private LocalDateTime createTime;

    @Schema(title = "更新时间")
    private LocalDateTime updateTime;

}

File: src/main/kotlin/plus/maa/backend/controller/response/copilotset/CopilotSetListRes.kt
package plus.maa.backend.controller.response.copilotset

import io.swagger.v3.oas.annotations.media.Schema
import plus.maa.backend.service.model.CopilotSetStatus
import java.time.LocalDateTime

/**
 * @author dragove
 * create on 2024-01-06
 */
@Schema(title = "作业集响应（列表）")
data class CopilotSetListRes (
    @Schema(title = "作业集id")
    val id: Long,

    @Schema(title = "作业集名称")
    val name: String,

    @Schema(title = "额外描述")
    val description: String,

    @Schema(title = "上传者id")
    val creatorId: String,

    @Schema(title = "上传者昵称")
    val creator: String,

    @Schema(title = "作业状态", enumAsRef = true)
    val status: CopilotSetStatus,

    @Schema(title = "创建时间")
    val createTime: LocalDateTime,

    @Schema(title = "更新时间")
    val updateTime: LocalDateTime,

    @Schema(title = "作业id列表")
    val copilotIds: List<Long>
)
